cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -223,13 +224,32 @@ public void eventReceived(Action action, KafkaRebalance kafkaRebalance) {
                     Reconciliation reconciliation = new Reconciliation("kafkarebalance-watch", kafkaRebalance.getKind(),
                             kafkaRebalance.getMetadata().getNamespace(), kafkaRebalance.getMetadata().getName());
 
-                    LOGGER.debugCr(reconciliation, "EventReceived {} on {} with status [{}] and {}={}", action,
+                    LOGGER.infoCr(reconciliation, "EventReceived {} on {} with status [{}] and {}={}", action,
                             kafkaRebalance.getMetadata().getName(),
                             kafkaRebalance.getStatus() != null ? rebalanceStateConditionType(kafkaRebalance.getStatus()) : null,
                             ANNO_STRIMZI_IO_REBALANCE, rawRebalanceAnnotation(kafkaRebalance));
 
-                    withLock(reconciliation, LOCK_TIMEOUT_MS,
-                        () -> reconcileRebalance(reconciliation, action == Action.DELETED ? null : kafkaRebalance));
+                    if (action == Action.MODIFIED) {
+                        KafkaRebalanceBuilder patchedKafkaRebalance = new KafkaRebalanceBuilder(kafkaRebalance);
+
+                        if (!kafkaRebalance.getMetadata().getAnnotations().containsKey(ANNO_KUBE_GENERATION)) {
+                            patchedKafkaRebalance.editMetadata().addToAnnotations(Map.of(ANNO_KUBE_GENERATION, "1")).endMetadata();
+                        }
+
+                        if (!patchedKafkaRebalance.buildMetadata().getAnnotations().get(ANNO_KUBE_GENERATION).equals(kafkaRebalance.getMetadata().getGeneration().toString())) {
+                            patchedKafkaRebalance.editMetadata()
+                                    .addToAnnotations(Map.of(ANNO_STRIMZI_IO_REBALANCE, KafkaRebalanceAnnotation.refresh.toString()))
+                                    .addToAnnotations(Map.of(ANNO_KUBE_GENERATION, kafkaRebalance.getMetadata().getGeneration().toString()))
+                                    .endMetadata();
```

Comment:
I'm not an expert on this. But this seems a bit weird and this can cause fights with any tooling managing the KafkaRebalance. Why don't you store the generation in the observed generation field in the status? Do you really need to self-annotate the resource to refresh it?

Reply:
Sure, Iam testing this at the moment. I guess it should work fine

Reply:
Works fine